indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -636,17 +644,31 @@ def _process(self):
         return jsonify_data(**self.list_generator.render_list())
 
 
-class RHRegistrationsModifyStatus(RHRegistrationsActionBase):
-    """Accept/Reject selected registrations."""
+class RHRegistrationsApprove(RHRegistrationsActionBase):
+    """Accept selected registrations from registration list."""
 
     def _process(self):
-        approve = request.form['flag'] == '1'
         for registration in self.registrations:
-            _modify_registration_status(registration, approve)
-        flash(_("The status of the selected registrations was updated successfully."), 'success')
+            _modify_registration_status(registration, approve=True)
+        flash(_("The selected registrations was successfully '{}'.").format('approved'), 'success')
         return jsonify_data(**self.list_generator.render_list())
 
 
+class RHRegistrationsReject(RHRegistrationsActionBase):
+    """Reject selected registrations from registration list."""
+
+    def _process(self):
+        form = RejectRegistrantsForm(obj=FormDefaults(registration_id=request.args.getlist('registration_id')))
+        message = _("Rejecting the selected registrations will trigger a notification email for each registrant.")
+        if form.validate_on_submit():
+            for registration in self.registrations:
+                registration.rejection_reason = form.rejection_reason.data
+                _modify_registration_status(registration, approve=False)
+            flash(_("The selected registrations was successfully '{}'.").format('rejected'), 'success')
```

Comment:
same here, don't use format on translated strings when the interpolated data is an english word (which needs to be translated btw) as well:

```suggestion
            flash(_("The selected registrations were successfully rejected."), 'success')
```